src/pudl/transform/ferc1.py

```diff
@@ -7,1400 +7,763 @@
+        fuel is just Uranium, or a mix of Uranium and Plutonium from decommissioned
+        nuclear weapons. See:
+
+        https://world-nuclear.org/information-library/facts-and-figures/heat-values-of-various-fuels.aspx
+
+        """
+        df = df.copy()
+
+        FuelFix = namedtuple("FuelFix", "fuel from_unit to_unit mult")
+        fuel_fixes = [
+            # US average coal heat content is 19.85 mmbtu/short ton
+            FuelFix("coal", "mmbtu", "ton", (1.0 / 19.85)),
+            FuelFix("coal", "btu", "ton", (1.0 / 19.85e6)),
+            # 2000 lbs per short ton
+            FuelFix("coal", "lbs", "ton", (1.0 / 2000.0)),
+            FuelFix("coal", "klbs", "ton", (1.0 / 2.0)),
+            # 42 gallons per barrel. Seriously, who makes up these units?
+            FuelFix("oil", "gal", "bbl", (1.0 / 42.0)),
+            FuelFix("oil", "kgal", "bbl", (1000.0 / 42.0)),
+            # On average a "ton of oil equivalent" is 7.46 barrels
+            FuelFix("oil", "ton", "bbl", 7.46),
+            FuelFix("gas", "mmbtu", "mcf", (1.0 / 1.037)),
+            # Nuclear plants report either heat content or mass of heavy metal
+            # MW*days thermal to MWh thermal
+            FuelFix("nuclear", "mwdth", "mwhth", 24.0),
+            # Straight energy equivalence between BTU and MWh here:
+            FuelFix("nuclear", "mmmbtu", "mwhth", (1.0 / 3.412142)),
+            FuelFix("nuclear", "btu", "mwhth", (1.0 / 3412142)),
+            # Unclear if it's possible to convert heavy metal to heat reliably
+            FuelFix("nuclear", "grams", "kg", (1.0 / 1000)),
+        ]
+        for fix in fuel_fixes:
+            fuel_mask = df.fuel_type_code_pudl == fix.fuel
+            unit_mask = df.fuel_units == fix.from_unit
+            df.loc[(fuel_mask & unit_mask), "fuel_consumed_units"] *= fix.mult
+            df.loc[(fuel_mask & unit_mask), "fuel_cost_per_unit_burned"] /= fix.mult
+            df.loc[(fuel_mask & unit_mask), "fuel_cost_per_unit_delivered"] /= fix.mult
+            df.loc[(fuel_mask & unit_mask), "fuel_units"] = fix.to_unit
+
+        # Set all remaining non-standard units and affected columns to NA.
+        FuelAllowedUnits = namedtuple("FuelAllowedUnits", "fuel allowed_units")
+        fuel_allowed_units = [
+            FuelAllowedUnits("coal", ("ton",)),
+            FuelAllowedUnits("oil", ("bbl",)),
+            FuelAllowedUnits("gas", ("mcf",)),
+            FuelAllowedUnits("nuclear", ("kg", "mwhth")),
+            FuelAllowedUnits("waste", ("ton",)),
+            # All unidentified fuel types ("other") get units set to NA
+            FuelAllowedUnits("other", ()),
+        ]
+        physical_units_cols = [
+            "fuel_consumed_units",
+            "fuel_cost_per_unit_burned",
+            "fuel_cost_per_unit_delivered",
+        ]
+        for fau in fuel_allowed_units:
+            fuel_mask = df.fuel_type_code_pudl == fau.fuel
+            unit_mask = ~df.fuel_units.isin(fau.allowed_units)
```

Comment:
for clarity, it'd be helpful imo to name this `non_allowed_unit_mask`

Reply:
Okay I renamed it to `invalid_unit_mask` to clarify it's BAD units (bad, bad units)